### PR TITLE
Make file style to be colored with same structure while color_only #405

### PR DESCRIPTION
**Pull request by ulwlu**
This pr changes below.

- Make file style to be colored with same structure while color_only

ss 4

ss 6

I confirmed below apart from test.

- `git -c 'interactive.diffFilter="./target/release/delta" --color-only' add -p` still works.

Remarks.

- Related to #405 
- I still need to resolve `commit-style` and `hunk-header-style`.
- Already ready for pr of `commit-style` and `hunk-header-style`, but they depend on this pr.

### CODE REVIEW

File: src/delta.rs [hunk 1]
```diff
@@ -180,7 +188,7 @@ where
             continue;
         }
 
-        if state == State::FileMeta && should_handle(&State::FileMeta, config) {
+        if state == State::FileMeta && should_handle(&State::FileMeta, config) && !config.color_only {
```
Comment (ulwlu) [inline src/delta.rs]: I think this can be refactored after `commit style` and `hunk header style` are also resolved.

File: src/delta.rs [hunk 2]
```diff
@@ -317,7 +325,7 @@ fn handle_generic_file_meta_header_line(
     raw_line: &str,
     config: &Config,
 ) -> std::io::Result<()> {
-    if config.file_style.is_omitted {
+    if config.file_style.is_omitted && !config.color_only {
         return Ok(());
     }
```
Comment (ulwlu) [inline src/delta.rs:331]: This requires anyway, otherwise we need to refactor `should_handle`.
We can discuss and make another pr if we need.

File: src/options/set.rs [hunk 3]
```diff
@@ -195,7 +195,6 @@ pub fn set_options(
         opt.side_by_side = false;
         opt.file_decoration_style = "none".to_string();
```
Comment (ulwlu) [inline src/options/set.rs:196]: This is required.
file_decoration_style will break structures anyway. We should prevent any of them.

**Comment by dandavison** [comment]
Great! Thanks for working on this @ulwlu.

> Already ready for pr of commit-style and hunk-header-style, but they depend on this pr.

You could open a second PR and use the PR target branch selector so that the second PR targets this branch. Unless I imagined it, I think GitHub is clever enough nowadays that the second one automatically switches to target master when the first merges.

File: src/delta.rs [hunk 4]
```diff
@@ -127,6 +131,10 @@ where
                 &plus_file,
             ));
             current_file_pair = Some((minus_file.clone(), plus_file.clone()));
+            if config.color_only {
+                handle_generic_file_meta_header_line(&mut painter, &line, &raw_line, config)?;
+                continue;
+            }
```
Comment (dandavison) [inline src/delta.rs:137]: Would this be clearer as an `else if`?

**Review by dandavison — CHANGES_REQUESTED** [review]
This is great, there's just one thing I'd ask for -- do you think you could add a failing test that demonstrates the problem that you are fixing? If you don't mind rebasing, it's nice to have the failing test be the first commit and the fixes come after (then I probably won't use squash-and-merge because the history is useful.)

The utility `ansi_test_utils::assert_line_has_style` might be useful for the test.

File: src/delta.rs (hunk 2 above)
Comment (dandavison) [reply src/delta.rs:331]: OK. Feel free to add explanatory comments to the code while it is fresh in your brain. If we have some test coverage of this then we'll feel even better about refactoring :)

Comment (ulwlu) [reply src/delta.rs:331]: Oh yes! I'm gonna add.

**Comment by ulwlu** [comment]
@dandavison 
Sure!

> You could open a second PR and use the PR target branch selector so that the second PR targets this branch. Unless I imagined it

Right, but I was not sure if this approach will be changed, and need to fix 2nd branch entirely or fix conflict (and I don't want so).

> do you think you could add a failing test that demonstrates the problem that you are fixing?

Sure, just a minute.

File: src/delta.rs (hunk 2 above)
Comment (ulwlu) [reply src/delta.rs:332]: @dandavison 
Add comment here def9554

**Comment by ulwlu** [comment]
@dandavison 

Add test here.
b48fa01

confirmed it's succeed now, and failed before the first commit.

**Comment by dandavison** [comment]
Perfect, thanks! Merged.
